src/Chapter2.hs

```diff
@@ -349,7 +351,8 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = subList 0 ((div (length l) 2) - 1) l
```

Comment:
Your implementation is correct, and nice usage of `subList`! However, `subList` does more than you need here, and it's enough to just _take_ the first half